bootstrap_cfn/fab_tasks.py

```diff
@@ -193,6 +193,94 @@ def swap_tags(tag1, tag2):
     r53_conn.update_dns_record(zone_id, fqdn2, 'TXT', '"{0}"'.format(stack_suffix1))
 
 
+@task
+def enter_maintenance(maintenance_ip):
+    """
+    Puts stack into maintenance mode
+
+    Sets all internet facing hostnames to resolve to given maintenance_ip
+    """
+    cfn_config = get_config()
+    r53_conn = get_connection(R53)
+
+    cached_zone_ids = {}
+    for elb in cfn_config.data['elb']:
+        if not elb['scheme'] == 'internet-facing':
+            continue
+
+        record = "{0}.{1}".format(elb['name'], elb['hosted_zone'])
+        zone_id = get_cached_zone_id(r53_conn, cached_zone_ids, elb['hosted_zone'])
+        r53_conn.update_dns_record(zone_id, record, 'A', maintenance_ip)
+
+
+@task
+def exit_maintenance():
+    """
+    Exit maintenance mode
+
+    Sets internet-facing elbs hostnames
+    back to the ELB DNS alias
+    """
+    r53_conn = get_connection(R53)
+    elb_conn = get_connection(ELB)
+
+    cfn_config = get_config()
+    stack_name = get_stack_name()
+
+    # get all elbs for this stack from AWS
+    stack_elbs = {}
+    [stack_elbs.update({
+                       getattr(x, 'logical_resource_id'): getattr(x, 'physical_resource_id')
+                       }) for x in elb_conn.cfn.get_stack_load_balancers(stack_name)]
```

Comment:
Rather than creating the dict and then updating it in a list comprehension (which is bad form, btw - this should just be a `for` loop), you can construct the dict with a list comprehension directly. Additionally, I would not use `getattr` here, as it's unnecessarily verbose:
```python
stack_elbs = dict([
    (x.logical_resource_id, x.physical_resource_id)
    for x in elb_conn.cfn.get_stack_load_balancers(stack_name)])
```